### PR TITLE
Add way to global handle Exceptions in event()

### PR DESCRIPTION
**Pull request by greenhost87**

### CODE REVIEW

File: modules/korolev/src/main/scala/korolev/server/KorolevServiceConfig.scala [hunk 1]
```diff
@@ -39,8 +39,12 @@ case class KorolevServiceConfig[F[_]: Effect, S, M](
   extensions: List[Extension[F, S, M]] = Nil,
   idGenerator: IdGenerator[F] = IdGenerator.default[F](),
   heartbeatInterval: FiniteDuration = 5.seconds,
-  reporter: Reporter = Reporter.PrintReporter
-)(implicit val executionContext: ExecutionContext)
+  reporter: Reporter = Reporter.PrintReporter,
+)(implicit val executionContext: ExecutionContext) {
+  def eventRecovery: Context.Access[F, _, _] => PartialFunction[Throwable, F[Unit]] = { _ =>
```
Comment (fomkin) [inline modules/korolev/src/main/scala/korolev/server/KorolevServiceConfig.scala:44]: How to define it? Extend KorolevServiceConfig?

File: modules/korolev/src/test/scala/korolev/Issue14Spec.scala [hunk 2]
```diff
@@ -56,7 +56,8 @@ class Issue14Spec extends AnyFlatSpec with Matchers {
         def setStyle(name: String, value: String): Unit = rc.setStyle(name, value)
         def addTextNode(text: String): Unit = rc.addTextNode(text)
         def addMisc(misc: Context.Binding[Future, Issue14Spec.S, Any]): Unit = k(rc, misc)
-      }
+      },
+      eventRecovery = ???
```
Comment (fomkin) [inline modules/korolev/src/test/scala/korolev/Issue14Spec.scala:60]: PartialFunction.empty

**Review by fomkin — COMMENTED** [review]
It's will be good to add recovery to render.

File: modules/korolev/src/main/scala/korolev/server/KorolevServiceConfig.scala (hunk 1 above)
Comment (greenhost87) [reply modules/korolev/src/main/scala/korolev/server/KorolevServiceConfig.scala:44]: yes

Comment (greenhost87) [reply modules/korolev/src/main/scala/korolev/server/KorolevServiceConfig.scala]: Change to parameter, no extend needed

File: modules/korolev/src/main/scala/korolev/internal/ComponentInstance.scala [hunk 3]
```diff
@@ -222,64 +224,69 @@ final class ComponentInstance
...
-    node(proxy)
+    try {
+      // Reset all event handlers delays and elements
+      prepare()
+      val state = snapshot[CS](nodeId).getOrElse(component.initialState)
+      val node =
+        try {
+          component.render(parameters, state)
+        } catch {
+          case e: MatchError =>
```
Comment (fomkin) [inline modules/korolev/src/main/scala/korolev/internal/ComponentInstance.scala:235]: Why not to move this case to catch section below?

File: modules/korolev/src/main/scala/korolev/internal/ComponentInstance.scala [hunk 4]
```diff
@@ -16,21 +16,22 @@
 
 package korolev.internal
 
-import korolev._
```
Comment (fomkin) [inline modules/korolev/src/main/scala/korolev/internal/ComponentInstance.scala:19]: Did you try to compile this with 2.12? Does it work?

File: modules/korolev/src/main/scala/korolev/server/KorolevServiceConfig.scala [hunk 5]
```diff
@@ -39,7 +39,10 @@ case class KorolevServiceConfig[F[_]: Effect, S, M](
   extensions: List[Extension[F, S, M]] = Nil,
   idGenerator: IdGenerator[F] = IdGenerator.default[F](),
   heartbeatInterval: FiniteDuration = 5.seconds,
-  reporter: Reporter = Reporter.PrintReporter
+  reporter: Reporter = Reporter.PrintReporter,
+  actionRecovery: Context.Access[F, _, _] => PartialFunction[Throwable, F[Unit]] = { _: Context.Access[F, _, _] =>
```
Comment (fomkin) [inline modules/korolev/src/main/scala/korolev/server/KorolevServiceConfig.scala:43]: stateLoader, document, reproter. Maybe recover or recovery?

File: modules/korolev/src/main/scala/korolev/internal/ComponentInstance.scala [hunk 6]
```diff
@@ -222,64 +224,69 @@ final class ComponentInstance
...
+                n.unsafeInitialize()
+                n.setEventsSubscription((e: Any) => entry.eventHandler(browserAccess, e).runAsyncForget)
+                n.applyRenderContext(entry.parameters, proxy, snapshot)
+            }
+        }
+      })
+      node(proxy)
+    } catch {
+      case NonFatal(ex) =>
+        actionRecovery(browserAccess)(ex)
```
Comment (fomkin) [inline modules/korolev/src/main/scala/korolev/internal/ComponentInstance.scala:288]: This will not work with Monix/IO/ZIO. You should invoke `runAsync`.

**Review by fomkin — CHANGES_REQUESTED** [review]

File: modules/korolev/src/main/scala/korolev/server/KorolevServiceConfig.scala (hunk 5 above)
Comment (greenhost87) [reply modules/korolev/src/main/scala/korolev/server/KorolevServiceConfig.scala:43]: changed to recovery

**Pull request merged**